examples/ppo/ppo_mujoco.py

```diff
@@ -0,0 +1,179 @@
+    from torchrl.envs import ExplorationType, set_exploration_type
+    from torchrl.objectives import ClipPPOLoss
+    from torchrl.objectives.value.advantages import GAE
+    from torchrl.record.loggers import generate_exp_name, get_logger
+    from utils_mujoco import make_env, make_ppo_models
+
+    # Define paper hyperparameters
+    device = "cpu" if not torch.cuda.is_available() else "cuda"
+    num_mini_batches = cfg.collector.frames_per_batch // cfg.loss.mini_batch_size
+    total_network_updates = (
+        (cfg.collector.total_frames // cfg.collector.frames_per_batch)
+        * cfg.loss.ppo_epochs
+        * num_mini_batches
+    )
+
+    # Create models (check utils_mujoco.py)
+    actor, critic = make_ppo_models(cfg.env.env_name)
+    actor, critic = actor.to(device), critic.to(device)
+
+    # Create collector
+    collector = SyncDataCollector(
+        create_env_fn=make_env(cfg.env.env_name, device),
+        policy=actor,
+        frames_per_batch=cfg.collector.frames_per_batch,
+        total_frames=cfg.collector.total_frames,
+        device=device,
+        storing_device=device,
+        max_frames_per_traj=-1,
+    )
+
+    # Create data buffer
+    sampler = SamplerWithoutReplacement()
+    data_buffer = TensorDictReplayBuffer(
+        storage=LazyMemmapStorage(cfg.collector.frames_per_batch, device=device),
+        sampler=sampler,
+        batch_size=cfg.loss.mini_batch_size,
+    )
+
+    # Create loss and adv modules
+    adv_module = GAE(
+        gamma=cfg.loss.gamma,
+        lmbda=cfg.loss.gae_lambda,
+        value_network=critic,
+        average_gae=False,
+    )
+    loss_module = ClipPPOLoss(
+        actor=actor,
+        critic=critic,
+        clip_epsilon=cfg.loss.clip_epsilon,
+        loss_critic_type=cfg.loss.loss_critic_type,
+        entropy_coef=cfg.loss.entropy_coef,
+        critic_coef=cfg.loss.critic_coef,
+        normalize_advantage=True,
+    )
+
+    # Create optimizers
+    actor_optim = torch.optim.Adam(actor.parameters(), lr=cfg.optim.lr)
+    critic_optim = torch.optim.Adam(critic.parameters(), lr=cfg.optim.lr)
+
+    # Create logger
+    exp_name = generate_exp_name("PPO", f"{cfg.logger.exp_name}_{cfg.env.env_name}")
+    logger = get_logger(cfg.logger.backend, logger_name="ppo", experiment_name=exp_name)
+
+    # Create test environment
+    test_env = make_env(cfg.env.env_name, device)
+    test_env.eval()
+
+    # Main loop
+    collected_frames = 0
+    num_network_updates = 0
+    start_time = time.time()
+    pbar = tqdm.tqdm(total=cfg.collector.total_frames)
+
+    for data in collector:
+
+        frames_in_batch = data.numel()
+        collected_frames += frames_in_batch
+        pbar.update(data.numel())
+
+        # Train loging
+        episode_rewards = data["next", "episode_reward"][data["next", "done"]]
+        if len(episode_rewards) > 0:
+            logger.log_scalar(
+                "reward_train", episode_rewards.mean().item(), collected_frames
```

Comment:
i usually use names like "train/reward", this will make wandb automatically divide them into different panels.

here is the stuff i log for training
https://github.com/pytorch/rl/blob/147de71d090d5705182bfabd24a99f3b2ee4cec9/examples/multiagent/utils/logging.py#L76

and this some for eval
https://github.com/pytorch/rl/blob/147de71d090d5705182bfabd24a99f3b2ee4cec9/examples/multiagent/utils/logging.py#L122

Reply:
i think we at least need to time the scripts and log times both for collection and training

Reply:
Makes sense to use this kind of naming. Probably in each example the metrics will vary a bit, but at least we can agree to make them always "train/..." and "eval/...".

Adding the timing also makes sense